django/contrib/postgres/forms/jsonb.py

```diff
@@ -52,3 +52,14 @@ def prepare_value(self, value):
         if isinstance(value, InvalidJSONInput):
             return value
         return json.dumps(value)
+
+    def has_changed(self, initial, data):
+        if super().has_changed(initial, data):
+            return True
+        data = self.to_python(data)
+        initial_value = initial if initial is not None else ''
+        data_value = data if data is not None else ''
+        # When value in JsonField's item changed from True to 1, or something like that,
+        # for example: {'a': True} -> '{"a": 1}',
+        # they need change to string using json.dumps to comparison.
```

Comment:
I think the document here is not very good, I will impove it later. Just leave s note here